glusterd2/commands/volumes/volume-option.go

```diff
@@ -175,6 +207,12 @@ func volumeOptionsHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	//save volume information for transaction failure scenario
+	if err := txn.Ctx.Set("oldvolinfo", volinfo); err != nil {
+		logger.WithError(err).Error("failed to set oldvolinfo in transaction context")
```

Comment:
You don't need the log here. It's logged at `Error` log level inside 

```go
func (c *Tctx) Set(key string, value interface{}) error {

        b, err := json.Marshal(value)
        if err != nil {
                c.logger.WithError(err).WithField("key", key).Error("failed to marshal value")
                return err
        }
```

Reply:
We can :+1: Feel free to do so.

Usually the caller should (optionally) log and inner function shouldn't. That's a good general practice. But in this specific case, it's beneficial to do it once internally because:

* the pattern repeats several times and these double logs clutter the code very much
* developers may forget to log in the caller
* developers may not use the right structured log format to log
* what is logged inside the function is same as what is logged outside - including the log level